cmd/ecr-credential-provider/main_test.go

```diff
@@ -24,24 +24,50 @@ import (
 	"testing"
 	"time"
 
-	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/ecr"
-	"github.com/aws/aws-sdk-go/service/ecrpublic"
-	"github.com/golang/mock/gomock"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/aws/aws-sdk-go-v2/service/ecr/types"
+	"github.com/aws/aws-sdk-go-v2/service/ecrpublic"
+	publictypes "github.com/aws/aws-sdk-go-v2/service/ecrpublic/types"
+	"github.com/stretchr/testify/mock"
```

Comment:
why switch the mocking lib?

Reply:
We only use the mocking library for ECR, and it's no longer supported (for a while now), so I decided just to kill it rather than figure out how to get everything updated. This puts it more in line with what we're doing everywhere else.

Reply:
We don't use ECR anything, or I might have pulled it out into it's own file. For this case, I figured it was cleaner to just inline it in the test file.

Reply:
Gotcha, didn't realize golang/mock had been put to pasture